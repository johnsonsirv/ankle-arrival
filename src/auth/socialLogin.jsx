import React from 'react';

const SocialLogin = () => {
  return <></>;
};

export default SocialLogin;
