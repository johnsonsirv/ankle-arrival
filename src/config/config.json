{
  "apiEndPoint": "ankle-arrival.herokuapp.com/api/v1",
  "demoApiEndPoint": "https://my-json-server.typicode.com/johnsonsirv/fake-json-api"
}
