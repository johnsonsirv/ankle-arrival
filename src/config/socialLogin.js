export const google = {
  appId: process.env.REACT_APP_GOOGLE_SOCIAL_LOGIN_APP_ID,
  providerName: 'google',
};

export const facebook = {
  appId: process.env.REACT_APP_FACEBOOK_SOCIAL_LOGIN_APP_ID,
  providerName: 'facebook',
};

export const twitter = {
  appId: process.env.REACT_APP_TWITTER_SOCIAL_LOGIN_APP_ID,
  providerName: 'twitter',
};

export const instagram = {
  appId: process.env.REACT_APP_INSTAGRAM_SOCIAL_LOGIN_APP_ID,
  providerName: 'instagram',
};
