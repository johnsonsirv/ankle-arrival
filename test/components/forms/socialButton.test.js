/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import SocialButton from '../../../src/components/forms/inputs/socialButton';
import { google } from '../../../src/config/socialLogin';

function setup() {
  const props = {
    value: 'Login with Google',
    provider: google.providerName,
    appId: google.appId,
    triggerLogin: jest.fn(),
  };
  const enzymeWrapper = mount(<SocialButton {...props} />);

  return { props, enzymeWrapper };
}

describe('<SocialButton /> rendering', () => {
  const { enzymeWrapper } = setup();
  it('renders correctly', () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });
  it('should render 1 <button>', () => {
    expect(enzymeWrapper.find('button')).toHaveLength(1);
  });
  it('should set a text value for button', () => {
    expect(enzymeWrapper.find('button').text()).toMatch(/Login with Google/);
  });
  it('should render <button> as enabled', () => {
    expect(enzymeWrapper.find('button').prop('disabled')).toBeUndefined();
  });
});

describe('<SocialButton /> interactions', () => {
  const { props, enzymeWrapper } = setup();
  it('should handle onClick', () => {
    enzymeWrapper.simulate('click', { target: {}, preventDefault: jest.fn() });
    expect(props.triggerLogin).toHaveBeenCalled();
  });
});
