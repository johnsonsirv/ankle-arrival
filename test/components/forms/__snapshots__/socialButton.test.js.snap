// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialButton /> rendering renders correctly 1`] = `ReactWrapper {}`;
